cmd/skopeo/inspect.go

```diff
@@ -235,3 +239,51 @@ func (opts *inspectOptions) writeOutput(stdout io.Writer, data any) error {
 	defer rpt.Flush()
 	return rpt.Execute([]any{data})
 }
+
+func manifestDigestFromManifest(manifestBlob []byte, img types.Image, userAlgorithm digest.Algorithm) (digest.Digest, error) {
+	if userAlgorithm != "" {
+		if !userAlgorithm.Available() {
+			return "", fmt.Errorf("digest algorithm %q is not available", userAlgorithm)
+		}
+		return userAlgorithm.FromBytes(manifestBlob), nil
+	}
+
+	configInfo := img.ConfigInfo()
+	if configInfo.Digest != "" {
+		alg := configInfo.Digest.Algorithm()
+		if alg.Available() {
+			return alg.FromBytes(manifestBlob), nil
+		}
+	}
+
+	return manifest.Digest(manifestBlob)
+}
+
+type algorithmValue digest.Algorithm
+
+func newAlgorithmValue(alg *digest.Algorithm) *algorithmValue {
+	return (*algorithmValue)(alg)
+}
+
+func (a *algorithmValue) Set(value string) error {
+	algorithm := digest.Algorithm(value)
+
+	if err := supporteddigests.TmpSetDigestForNewObjects(algorithm); err != nil {
```

Comment:
I think there is a conceptual difference between _parsing_ options and _handling_ options; causing side effects at the parsing stage is rather unexpected. So if the `TmpSet…` call should be made, I’d prefer that to happen later in `inspectOptions.run`.

(For the parsing, `digest.Algorithm` already implements `Set` but not `Type`. That implementation of `Set` also hard-codes that `""` means sha256… for now that works.)